File: baekjoon_python/1094.py
"""

1094번

입력한 값을 2진법으로 바꾼 후 1의 개수도 정답

print(bin(int(input())).count('1'))

"""


x = int(input())
stick = [64]
while sum(stick) > x:
    stick = [stick[0] / 2] * 2 + stick[1:]
    if sum(stick[1:]) >= x:
        stick = stick[1:]
print(len(stick))

File: baekjoon_kotlin/_1000/1094.kt
package baekjoon_kotlin._1000/*

1094번 - 막대기
https://www.acmicpc.net/problem/1094

배열을 사용해서 값을 추가하며 판별하거나, 여러 막대를 자르지 않기 때문에 변수 3개만 이용해도 된다.

*/

fun main() {
    val x = System.`in`.bufferedReader().readLine().toInt()
    var count = 1 // 막대의 개수
    var stick = 64 // 막대의 총 길이
    var last = 64 // 잘라진 막대의 길이
    while (stick > x) {
        last /= 2
        if (stick - last < x) count += 1
        else stick -= last
    }
    print(count)
}

/*

*/